Example_Card_Calypso/src/main/java/org/eclipse/keyple/card/calypso/example/UseCase12_PerformanceMeasurement_Validation/Main_PerformanceMeasurement_Validation_Pcsc.java

```diff
@@ -0,0 +1,286 @@
+    // Select the card and read the record 1 of the file ENVIRONMENT_AND_HOLDER
+    CalypsoCardSelection cardSelection =
+        calypsoCardService
+            .createCardSelection()
+            .acceptInvalidatedCard()
+            .filterByCardProtocol(ConfigurationUtil.ISO_CARD_PROTOCOL)
+            .filterByDfName(cardAid);
+
+    CardSecuritySetting cardSecuritySetting =
+        CalypsoExtensionService.getInstance()
+            .createCardSecuritySetting()
+            .setControlSamResource(samReader, calypsoSam);
+
+    boolean loop = true;
+    while (loop) {
+      logger.info(
+          "{}########################################################{}", ANSI_YELLOW, ANSI_RESET);
+      logger.info(
+          "{}## Press ENTER when the card is in the reader's field ##{}", ANSI_YELLOW, ANSI_RESET);
+      logger.info(
+          "{}## (or press 'q' + ENTER to exit)                     ##{}", ANSI_YELLOW, ANSI_RESET);
+      logger.info(
+          "{}########################################################{}", ANSI_YELLOW, ANSI_RESET);
+
+      String input = bufferedReader.readLine();
+
+      if (input.toLowerCase().contains("q")) {
+        bufferedReader.close();
+        loop = false;
+        continue;
+      }
+
+      try {
+        logger.info("Starting validation transaction...");
+
+        logger.info("Select application with AID = '{}'", cardAid);
+
+        // Prepare the selection by adding the created Calypso selection to the card selection
+        // scenario.
+        cardSelectionManager.prepareSelection(cardSelection);
+
+        CardSelectionResult cardSelectionResult =
+            cardSelectionManager.processCardSelectionScenario(cardReader);
+
+        CalypsoCard calypsoCard = (CalypsoCard) cardSelectionResult.getActiveSmartCard();
+
+        if (calypsoCard == null) {
+          logger.info("Card selection failed!");
+          continue;
+        }
+
+        // read the current time used later to compute the transaction time
+        long timeStamp = System.currentTimeMillis();
+        // create a transaction manager, open a Secure Session, read Environment and Event Log.
+        CardTransactionManager cardTransactionManager =
+            CalypsoExtensionService.getInstance()
+                .createCardTransaction(cardReader, calypsoCard, cardSecuritySetting)
+                .prepareReadRecord(
+                    CalypsoConstants.SFI_ENVIRONMENT_AND_HOLDER, CalypsoConstants.RECORD_NUMBER_1)
+                .prepareReadRecord(CalypsoConstants.SFI_EVENT_LOG, CalypsoConstants.RECORD_NUMBER_1)
+                .processOpening(WriteAccessLevel.DEBIT);
+
+        /*
+        Place for the analysis of the context and the last event log
+        */
+
+        // read the contract list
+        cardTransactionManager
+            .prepareReadRecord(CalypsoConstants.SFI_CONTRACT_LIST, CalypsoConstants.RECORD_NUMBER_1)
+            .processCommands();
+        /*
```

Comment:
To simulate that there is a minimum of terminal processing after **each** card read (including reads of the opening), I would at least assign the content of each reading to a local attribute, for example:

byte[] contractListData = calypsoCard.CalypsoCard.getFileBySfi(CalypsoConstants.SFI_CONTRACT_LIST).getData().getContent(CalypsoConstants.RECORD_NUMBER_1);